packages/main/types/server/caching.ts

```diff
@@ -0,0 +1,72 @@
+/**
+ * @license
+ * Copyright 2024 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+import { Content, Part } from "../content";
+import { Tool, ToolConfig } from "../function-calling";
+
+/**
+ * @public
+ */
+export interface CachedContentBase {
+  model?: string;
+  contents: Content[];
+  tools?: Tool[];
+  toolConfig?: ToolConfig;
+  systemInstruction?: string | Part | Content;
+}
+
+/**
+ * Describes CachedContent interface for sending to the server (if creating)
+ * or received from the server (using getters or list methods).
+ * @public
+ */
+export interface CachedContent extends CachedContentBase {
+  name?: string;
+  ttl?: string;
```

Comment:
Is there a particular format we should define here? Seconds, or something?

Reply:
The input (CachedContentCreateParams) takes `ttlSeconds`. This is the format of the JSON that we send/receive from the API endpoint.

Reply:
👍